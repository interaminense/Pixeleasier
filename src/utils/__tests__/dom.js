import {
	createElement,
	getElementByType,
	setStyle,
	setAttr
} from '../dom';

describe('createElement', () => {
	it('should return a div element created without any parameter', () => {
		const element = createElement();

		expect(element).toMatchSnapshot();
	});

	it('should return a div element created', () => {
		const element = createElement('div');

		expect(element).toMatchSnapshot();
	});

	it('should return a div element created with a classname', () => {
		const element = createElement('div', 'my-classname');

		expect(element).toMatchSnapshot();
	});

	it('should return a div element created with two classnames', () => {
		const element = createElement('div', ['my-classname', 'my-classname-2']);

		expect(element).toMatchSnapshot();
	});
});

describe('getElementByType', () => {
	it('should return a element type span', () => {
		const span = getElementByType('span');

		expect(span).toMatchSnapshot();
	});

	it('should return a element type h1', () => {
		const h1 = getElementByType('h1');

		expect(h1).toMatchSnapshot();
	});

	it('should return a element type svg', () => {
		const svg = getElementByType('svg');

		expect(svg).toMatchSnapshot();
	});

	it('should return a element type rect', () => {
		const rect = getElementByType('rect');

		expect(rect).toMatchSnapshot();
	});
});

describe('setStyle', () => {
	it('should return a div element with background-color red style', () => {
		const element = createElement('div');

		expect(setStyle(element, {backgroundColor: 'red'})).toMatchSnapshot();
	});

	it('should return a div element with more than one style', () => {
		const element = createElement('div');
		const styles = setStyle(element, {
			backgroundColor: 'red',
			color: 'green',
			fontSize: '14px',
			height: '100px',
			width: '100px',
		});

		expect(styles).toMatchSnapshot();
	});
});

describe('setAttr', () => {
	it('should return a div element with a title attribute', () => {
		const element = createElement('div');

		expect(setAttr(element, {title: 'my title'})).toMatchSnapshot();
	});

	it('should return a div element with more than one attribute', () => {
		const element = createElement('svg');
		const attrs = setAttr(element, {
			style: 'fill:red',
			width: '100px',
			height: '100px',
			x: '10px',
			y: '10px'
		});

		expect(attrs).toMatchSnapshot();
	});
});