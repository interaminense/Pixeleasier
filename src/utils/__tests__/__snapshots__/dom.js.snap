// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createElement should return a div element created 1`] = `<div />`;

exports[`createElement should return a div element created with a classname 1`] = `
<div
  class="my-classname"
/>
`;

exports[`createElement should return a div element created with two classnames 1`] = `
<div
  class="my-classname my-classname-2"
/>
`;

exports[`createElement should return a div element created without any parameter 1`] = `<div />`;

exports[`getElementByType should return a element type h1 1`] = `<h1 />`;

exports[`getElementByType should return a element type rect 1`] = `<rect />`;

exports[`getElementByType should return a element type span 1`] = `<span />`;

exports[`getElementByType should return a element type svg 1`] = `<svg />`;

exports[`setAttr should return a div element with a title attribute 1`] = `
<div
  title="my title"
/>
`;

exports[`setAttr should return a div element with more than one attribute 1`] = `
<svg
  height="100px"
  style="fill:red"
  width="100px"
  x="10px"
  y="10px"
/>
`;

exports[`setStyle should return a div element with background-color red style 1`] = `
<div
  style="background-color: red;"
/>
`;

exports[`setStyle should return a div element with more than one style 1`] = `
<div
  style="background-color: red; color: green; font-size: 14px; height: 100px; width: 100px;"
/>
`;
